File: src/utils/initialState/rooms.ts
import type { Room, ShipTypes } from '../../types/ship'

export interface GetInitialShipRoomsParams {
  shipType: ShipTypes
}
export const GetInitialShipRooms = (
  { shipType }: GetInitialShipRoomsParams = { shipType: 'griffin' },
): Map<string, Room> => {
  const rooms = new Map<string, Room>()

  switch (shipType) {
    case 'griffin':
      rooms.set('room-1', { type: 'bridge', name: 'Bridge', broken: [] })
      rooms.set('room-2', {
        type: 'garden',
        name: 'Hydroponic Garden',
        broken: [],
      })
      rooms.set('room-3', { type: 'refectory', name: 'Refectory', broken: [] })
      rooms.set('room-4', { type: 'rnd', name: 'R&D', broken: [] })
      rooms.set('room-5', { type: 'medlab', name: 'Medical Lab', broken: [] })
      rooms.set('room-6', {
        type: 'life-support',
        name: 'Life Support System',
        broken: [],
      })
      rooms.set('room-7', { type: 'storage', name: 'Storage Room', broken: [] })
      rooms.set('room-8', {
        type: 'private-quarters',
        name: 'Private Quarters',
        broken: [],
      })
      rooms.set('room-9', {
        type: 'common-area',
        name: 'Common Area',
        broken: [],
      })
      rooms.set('room-10', { type: 'dorms', name: 'Dorms', broken: [] })
      rooms.set('room-11', { type: 'barracks', name: 'Barracks', broken: [] })
      rooms.set('room-12', {
        type: 'weapon-systems',
        name: 'Weapon Systems',
        broken: [],
      })
      rooms.set('room-13', {
        type: 'fightercrafts-bay',
        name: 'Buzzards Bay',
        broken: [],
      })
      rooms.set('room-14', {
        type: 'expocrafts-bay',
        name: 'Blitzhoppers Bay',
        broken: [],
      })
      rooms.set('room-15', {
        type: 'flak-turret',
        name: 'Front Flak Turret',
        broken: [],
      })
      rooms.set('room-16', {
        type: 'laser-turret',
        name: 'Rear Laser Turret',
        broken: [],
      })
      rooms.set('room-17', { type: 'armory', name: 'Armory', broken: [] })
      rooms.set('room-18', {
        type: 'engine-room',
        name: 'Engine Room',
        broken: [],
      })
      rooms.set('room-19', {
        type: 'corridor',
        name: 'Front Corridor',
        broken: [],
      })
      rooms.set('room-20', {
        type: 'hallway',
        name: 'Front Hallway',
        broken: [],
      })
      rooms.set('room-21', {
        type: 'corridor',
        name: 'Rear Corridor',
        broken: [],
      })
      rooms.set('room-22', {
        type: 'hallway',
        name: 'Rear Hallway',
        broken: [],
      })
      rooms.set('room-23', {
        type: 'plating',
        name: "Griffin's Plating 1",
        broken: [],
      })
      rooms.set('room-24', {
        type: 'plating',
        name: "Griffin's Plating 2",
        broken: [],
      })
      rooms.set('room-25', {
        type: 'plating',
        name: "Griffin's Plating 3",
        broken: [],
      })
      rooms.set('room-26', {
        type: 'plating',
        name: "Griffin's Plating 4",
        broken: [],
      })
      rooms.set('room-27', {
        type: 'plating',
        name: "Griffin's Plating 5",
        broken: [],
      })
      rooms.set('room-28', {
        type: 'plating',
        name: "Griffin's Plating 6",
        broken: [],
      })
      rooms.set('room-29', {
        type: 'plating',
        name: "Griffin's Plating 7",
        broken: [],
      })
      rooms.set('room-30', {
        type: 'plating',
        name: "Griffin's Plating 8",
        broken: [],
      })
      break
  }

  return rooms
}

File: src/nakama/gamelogic/utils/initial/rooms.go
package initial

import "nanosh/nakama-modules/gamelogic/types"

// GetInitialShipRoomsParams defines parameters for getting initial ship rooms
type GetInitialShipRoomsParams struct {
	ShipType types.ShipType `json:"shipType"`
}

// GetInitialShipRooms creates and returns a map of all rooms for a given ship type
// Defaults to griffin if no ship type provided
func GetInitialShipRooms(params *GetInitialShipRoomsParams) map[string]*types.Room {
	// Default to griffin if no params provided
	if params == nil {
		params = &GetInitialShipRoomsParams{ShipType: types.ShipGriffin}
	}

	rooms := make(map[string]*types.Room)

	switch params.ShipType {
	case types.ShipGriffin:
		rooms["room-1"] = &types.Room{
			Type: types.RoomBridge,
			Name: "Bridge",
		}
		rooms["room-2"] = &types.Room{
			Type: types.RoomGarden,
			Name: "Hydroponic Garden",
		}
		rooms["room-3"] = &types.Room{
			Type: types.RoomRefectory,
			Name: "Refectory",
		}
		rooms["room-4"] = &types.Room{
			Type: types.RoomRND,
			Name: "R&D",
		}
		rooms["room-5"] = &types.Room{
			Type: types.RoomMedlab,
			Name: "Medical Lab",
		}
		rooms["room-6"] = &types.Room{
			Type: types.RoomLifeSupport,
			Name: "Life Support System",
		}
		rooms["room-7"] = &types.Room{
			Type: types.RoomStorage,
			Name: "Storage Room",
		}
		rooms["room-8"] = &types.Room{
			Type: types.RoomPrivateQuarters,
			Name: "Private Quarters",
		}
		rooms["room-9"] = &types.Room{
			Type: types.RoomCommonArea,
			Name: "Common Area",
		}
		rooms["room-10"] = &types.Room{
			Type: types.RoomDorms,
			Name: "Dorms",
		}
		rooms["room-11"] = &types.Room{
			Type: types.RoomBarracks,
			Name: "Barracks",
		}
		rooms["room-12"] = &types.Room{
			Type: types.RoomWeaponSystems,
			Name: "Weapon Systems",
		}
		rooms["room-13"] = &types.Room{
			Type: types.RoomFighterCraftsBay,
			Name: "Buzzards Bay",
		}
		rooms["room-14"] = &types.Room{
			Type: types.RoomExpoCraftsBay,
			Name: "Blitzhoppers Bay",
		}
		rooms["room-15"] = &types.Room{
			Type: types.RoomFlakTurret,
			Name: "Front Flak Turret",
		}
		rooms["room-16"] = &types.Room{
			Type: types.RoomLaserTurret,
			Name: "Rear Laser Turret",
		}
		rooms["room-17"] = &types.Room{
			Type: types.RoomArmory,
			Name: "Armory",
		}
		rooms["room-18"] = &types.Room{
			Type: types.RoomEngineRoom,
			Name: "Engine Room",
		}
		rooms["room-19"] = &types.Room{
			Type: types.RoomCorridor,
			Name: "Front Corridor",
		}
		rooms["room-20"] = &types.Room{
			Type: types.RoomHallway,
			Name: "Front Hallway",
		}
		rooms["room-21"] = &types.Room{
			Type: types.RoomCorridor,
			Name: "Rear Corridor",
		}
		rooms["room-22"] = &types.Room{
			Type: types.RoomHallway,
			Name: "Rear Hallway",
		}
		rooms["room-23"] = &types.Room{
			Type: types.RoomPlating,
			Name: "Griffin's Plating 1",
		}
		rooms["room-24"] = &types.Room{
			Type: types.RoomPlating,
			Name: "Griffin's Plating 2",
		}
		rooms["room-25"] = &types.Room{
			Type: types.RoomPlating,
			Name: "Griffin's Plating 3",
		}
		rooms["room-26"] = &types.Room{
			Type: types.RoomPlating,
			Name: "Griffin's Plating 4",
		}
		rooms["room-27"] = &types.Room{
			Type: types.RoomPlating,
			Name: "Griffin's Plating 5",
		}
		rooms["room-28"] = &types.Room{
			Type: types.RoomPlating,
			Name: "Griffin's Plating 6",
		}
		rooms["room-29"] = &types.Room{
			Type: types.RoomPlating,
			Name: "Griffin's Plating 7",
		}
		rooms["room-30"] = &types.Room{
			Type: types.RoomPlating,
			Name: "Griffin's Plating 8",
		}
	}

	return rooms
}